Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -0,0 +1,380 @@
+
+            var folder = await GetCacheFolderAsync();
+            var files = await folder.GetFilesAsync();
+
+            var filesToDelete = new List<StorageFile>();
+
+            foreach (var file in files)
+            {
+                if (await IsFileOutOfDate(file, expirationDate).ConfigureAwait(false))
+                {
+                    filesToDelete.Add(file);
+                }
+            }
+
+            await InternalClearAsync(files).ConfigureAwait(false);
+
+            _inMemoryFileStorage.Clear(duration);
+        }
+
+        /// <summary>
+        /// Assures that image is available in the cache
+        /// </summary>
+        /// <param name="uri">Uri of the image</param>
+        /// <param name="fileName">fileName to for local storage</param>
+        /// <param name="throwOnError">Indicates whether or not exception should be thrown if imagge cannot be loaded</param>
+        /// <param name="storeToMemoryCache">Indicates if image should be available also in memory cache</param>
+        /// <returns>void</returns>
+        public Task PreCacheAsync(Uri uri, string fileName, bool throwOnError = false, bool storeToMemoryCache = false)
+        {
+            return GetItemAsync(uri, fileName, throwOnError, !storeToMemoryCache);
+        }
+
+        /// <summary>
+        /// Load a specific image from the cache. If the image is not in the cache, ImageCache will try to download and store it.
+        /// </summary>
+        /// <param name="uri">Uri of the image.</param>
+        /// <param name="fileName">fileName to for local storage</param>
+        /// <param name="throwOnError">Indicates whether or not exception should be thrown if imagge cannot be loaded</param>
+        /// <returns>a BitmapImage</returns>
+        public Task<T> GetFromCacheAsync(Uri uri, string fileName, bool throwOnError = false)
+        {
+            return GetItemAsync(uri, fileName, throwOnError, false);
+        }
+
+        /// <summary>
+        /// Cache specific hooks to proccess items from http response
+        /// </summary>
+        /// <param name="stream">input stream</param>
+        /// <returns>awaitable task</returns>
+        protected virtual async Task<T> InitializeTypeAsync(IRandomAccessStream stream)
+        {
+            // nothing to do in this instance;
+            return default(T);
+        }
+
+        /// <summary>
+        /// Cache specific hooks to proccess items from http response
+        /// </summary>
+        /// <param name="baseFile">storage file</param>
+        /// <returns>awaitable task</returns>
+        protected virtual async Task<T> InitializeTypeAsync(StorageFile baseFile)
+        {
+            // nothing to do in this instance;
+            return default(T);
+        }
+
+        private async Task<T> GetItemAsync(Uri uri, string fileName, bool throwOnError, bool preCacheOnly)
+        {
+            T t = default(T);
```

Comment:
Can we not use single letter variable names please. Looking at t. further down the code is not clear what t. is.